open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/validate/impl/DefaultConsentValidator.java

```diff
@@ -381,4 +383,116 @@ private static void validateFundsConfirmationSubmission(ConsentValidateData cons
+            return;
+        }
+
+        JSONObject initiationParameterValidationResults = VRPSubmissionPayloadValidator.
+                validateInitiationParameter(submissionData);
+        if (!Boolean.parseBoolean(initiationParameterValidationResults.
+                getAsString(ConsentExtensionConstants.IS_VALID_PAYLOAD))) {
+            log.error(initiationParameterValidationResults.getAsString(ConsentExtensionConstants.ERROR_MESSAGE));
+            ConsentValidatorUtil.getErrorAndLog(initiationParameterValidationResults, consentValidationResult);
+            return;
+        }
+
+        JSONObject instructionParameterValidationResults = VRPSubmissionPayloadValidator.
+                validateInstructionParameter(submissionData);
+        if (!Boolean.parseBoolean(instructionParameterValidationResults.
+                getAsString(ConsentExtensionConstants.IS_VALID_PAYLOAD))) {
+            log.error(instructionParameterValidationResults.getAsString(ConsentExtensionConstants.ERROR_MESSAGE));
+            ConsentValidatorUtil.getErrorAndLog(instructionParameterValidationResults, consentValidationResult);
+            return;
+        }
+
+        // Check if requested consent ID in the body to initiation consent ID.
+        if (!submissionData.containsKey(ConsentExtensionConstants.CONSENT_ID) ||
+                !(submissionData.get(ConsentExtensionConstants.CONSENT_ID) instanceof String) ||
+                !submissionData.get(ConsentExtensionConstants.CONSENT_ID)
+                        .equals(detailedConsentResource.getConsentID())) {
+            log.error(ErrorConstants.INVALID_REQUEST_CONSENT_ID);
+            consentValidationResult.setErrorMessage(ErrorConstants.INVALID_REQUEST_CONSENT_ID);
+            consentValidationResult.setErrorCode(ErrorConstants.RESOURCE_CONSENT_MISMATCH);
+            consentValidationResult.setHttpCode(HttpStatus.SC_BAD_REQUEST);
+            return;
+        }
+
+        JSONObject dataObject = (JSONObject) initiationJson.get(ConsentExtensionConstants.DATA);
+        JSONObject requestInitiation = (JSONObject) dataObject.get(ConsentExtensionConstants.INITIATION);
+        JSONObject submissionInitiation = (JSONObject) submissionData.get(ConsentExtensionConstants.INITIATION);
+        JSONObject submissionInstruction = (JSONObject) submissionData.get(ConsentExtensionConstants.INSTRUCTION);
+
+        JSONObject initiationValidationResult = VRPSubmissionPayloadValidator
+                .validateInitiation(submissionInitiation, requestInitiation);
+
+        if (!Boolean.parseBoolean(initiationValidationResult.
+                getAsString(ConsentExtensionConstants.IS_VALID_PAYLOAD))) {
+            ConsentValidatorUtil.getErrorAndLog(initiationValidationResult, consentValidationResult);
+            return;
+        }
+
+        JSONObject instructionValidationResult = VRPSubmissionPayloadValidator.
+                validateInstruction(submissionInstruction, requestInitiation);
```

Comment:
Here shouldn't we pass requestInstruction ? rather than requestInitiation?

Reply:
Here we consider the initiation parameter since the creditor account from the initiation request need to be retrieved. Method comments and relevant inline comments are been added. Fixed in [983db07]